providers/oidc.go

```diff
@@ -175,14 +176,24 @@ func (p *OIDCProvider) createSessionState(ctx context.Context, token *oauth2.Tok
 	return newSession, nil
 }
 
-func (p *OIDCProvider) CreateSessionStateFromBearerToken(ctx context.Context, rawIDToken string, idToken *oidc.IDToken) (*sessions.SessionState, error) {
+func (p *OIDCProvider) CreateSessionFromToken(ctx context.Context, token string, verify middleware.VerifyFunc) (*sessions.SessionState, error) {
```

Comment:
I wanted to ease the implementation arguments for this to make it ready for all providers (i.e. not needing two forms of the token that is unique to OIDC post oidc.verify).

At the moment, I think we need to inject in the verifierFunc to handle the generic case (with `extra-jwt-issuers`) where they use the single common function but have different oidc validators based on the `issuer=audience` pair.

Long term, when we potential make that just a chain of `bearer` providers that implement `CreateSessionFromToken` -- I think each provider can then store the verifier object in its attributes and we won't need to pass it in anymore.

At that point this method signature can be further simplified to `CreateSessionFromToken(ctx context.Context, token string)`.

We also then don't need to store the get `GetOIDCVerifier` method on `options`... which feels out of place even now and probably belongs on the Provider itself?

Reply:
Hmmmm... I could ditch the `verifyFunc` argument now (and move ownership of the OIDC verifier object directly into the OIDC provider).

Then for the generic case I can build distinct `middleware.TokenToSessionFunc` for the `TokenToSessionLoaders` that are closures around the needed OIDC verifier (rather than getting a default implementation that gets passed a distinct VerifyFunc).

This is tempting... 🤔